### PR TITLE
Lesson 4, Lexical Structure

### PR DESCRIPTION
**Pull request by santiagobasulto**

### CODE REVIEW

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+value = 5
+if value >= 5:
+    for i in range(5):
+        for j in range(5):
+            print((j, i))
+else:
+    print("Value too small. Block ignored.")
+```
+
+We have defined 3 blocks. The first and outermost is the one created by the `if` statement. If the value is not greater or equal than 5 the entire block will be ignored. Inside the `if` statement, in the _true branch_ we have defined 2 more blocks. Each block indents the code 4 spaces to the right. Here's an important note to remember: **always use 4 spaces to indent your code**. And more importantly **Don't use tab characters**. Tabs are valid in Python programs but they are [discouraged](https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces).
```
Comment (Soledadb) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: If the value is not greater or equal than 5, the entire...
Inside the `if` statement, in the _true branch_, we have defined

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+print("Ugh, I don't like semicolons");  # Don't use them
+```
+
+## Blank lines
+
+You can have as many blank lines as you want, but there are [good practices](https://www.python.org/dev/peps/pep-0008/#blank-lines) that we like to follow.
+
+## Character sets
+
+As we'll see in more detail later, Python 2 doesn't support Unicode. That limitation doesn't just raise an issue when dealing with strings programmatically, but also when we're _writing_ our programs. The only valid characters in a Python 2 string literals are ASCII characters. If you want to include non-ASCII characters you have to put include statement in the first line of your source file: `# -*- coding: utf-8 -*-`. Example:
```
Comment (Soledadb) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: If you want to include non-ASCII characters, you have to

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+```python
+# -*- coding: utf-8 -*-
+
+print("Non-ASCII chars: ñ à á ó ò")
+```
+
+If you remove the first comment line the program won't even compile and you'll get a `SyntaxError`. Fortunately that won't happen in Python 3, in which you can include any Unicode character in your String literals.
```
Comment (Soledadb) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: If you remove the first comment line, the program
Fortunately, that won't happen

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+-my_var = 'X'
+$var = 'R'
+4var = 'A'
+```
+
+There are keywords reserved to Python that can't be used as identifiers. You can see a complete list for Python 3 [here](https://docs.python.org/3/reference/lexical_analysis.html#keywords) and for Python 2 [here](https://docs.python.org/2/reference/lexical_analysis.html#keywords).
+
+## String literals
+
+Strings can be defined through the use of "String literals". To do so you just need to enclose the text you want in single or double quotes. Both are valid and they're used somehow interchangeably. Examples of Strings:
```
Comment (Soledadb) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: To do so, you just need to enclose

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md (hunk 1 above)
Comment (santiagobasulto) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: I'll leave "discouraged". It's a term commonly used in tech environments (maybe we're doing it wrong, but it'll sound familiar). Check this out from the official Python documentation ([https://docs.python.org/2/library/subprocess.html](https://docs.python.org/2/library/subprocess.html)): 

> For this reason, the use of shell=True is strongly discouraged in cases where the command string is constructed from external input

**Comment by santiagobasulto** [comment]
All fixed. Thanks @Soledadb

File: unit-1-python-introduction/lesson-4-lexical-structure/README.md (hunk 1 above)
Comment (Soledadb) [inline unit-1-python-introduction/lesson-4-lexical-structure/README.md]: Jaja Okay dale, besooo

Enviado desde mi iPhone

**Pull request merged**